src/core_plugins/kibana/public/dashboard/grid/dashboard_grid.js

```diff
@@ -108,8 +109,9 @@ export class DashboardGrid extends React.Component {
           i: panelLayout.i,
         }
       };
-      onPanelUpdated(updatedPanel);
+      updatedPanels.push(updatedPanel);
     });
+    onPanelsUpdated(updatedPanels);
```

Comment:
This is the meat of the fix - need to only emit a single update at the end of the function, not incrementally throughout.